File: ru.kmorozov.gbd.desktop/src/main/java/ru/kmorozov/gbd/core/config/LocalSystemOptions.java
package ru.kmorozov.gbd.core.config;

import ru.kmorozov.gbd.core.config.options.CtxOptions;
import ru.kmorozov.gbd.core.loader.LocalFSStorage;

/**
 * Created by km on 06.12.2015.
 */
public class LocalSystemOptions implements IGBDOptions {

    @Override
    public String getBookId() {
        return SystemConfigs.getLastBookId();
    }

    @Override
    public IStorage getStorage() {
        return new LocalFSStorage(SystemConfigs.getRootDir());
    }

    @Override
    public String getProxyListFile() {
        return SystemConfigs.getProxyListFile();
    }

    @Override
    public int getImageWidth() {
        return SystemConfigs.getResolution();
    }

    @Override
    public boolean reloadImages() {
        return SystemConfigs.getReload();
    }

    @Override
    public boolean secureMode() {
        return SystemConfigs.getSecureMode();
    }

    @Override
    public String pdfOptions() {
        return SystemConfigs.getPdfMode();
    }

    @Override
    public CtxOptions ctxOptions() {
        return CtxOptions.DEFAULT_CTX_OPTIONS;
    }
}

File: ru.kmorozov.gbd.desktop/src/main/java/ru/kmorozov/gbd/core/config/LocalSystemOptions.kt
package ru.kmorozov.gbd.core.config

import ru.kmorozov.gbd.core.config.options.CtxOptions
import ru.kmorozov.gbd.core.config.options.ScanOptions
import ru.kmorozov.gbd.core.loader.LocalFSStorage

/**
 * Created by km on 06.12.2015.
 */
class LocalSystemOptions : IGBDOptions {

    override val bookId: String
        get() = SystemConfigs.lastBookId

    override val storage: IStorage
        get() = LocalFSStorage.getStorage(SystemConfigs.rootDir)

    override val proxyListFile: String
        get() = SystemConfigs.proxyListFile

    override val imageWidth: Int
        get() = SystemConfigs.resolution

    override val reloadImages: Boolean
        get() = SystemConfigs.reloadImages

    override fun scanOptions(): ScanOptions {
        TODO("Not yet implemented")
    }

    override val secureMode: Boolean
        get() = SystemConfigs.secureMode


    override fun pdfOptions(): String {
        return SystemConfigs.pdfMode
    }

    override fun ctxOptions(): CtxOptions {
        return CtxOptions.DEFAULT_CTX_OPTIONS
    }
}
